### PR TITLE
Fix broken checks

### PR DESCRIPTION
**Pull request by jameswald**

### CODE REVIEW

File: src/idp/response_builder.rs [hunk 1]
```diff
@@ -39,7 +40,7 @@ fn build_authn_statement(class: AuthenticationContextClass) -> AuthnStatement {
 
 pub struct ResponseAttribute<'a> {
     pub required_attribute: RequiredAttribute,
-    pub values: &'a [&'a str],
+    pub values: Cow<'a, [&'a str]>,
```
Comment (jameswald) [inline src/idp/response_builder.rs:43]: May need a `'b` lifetime here too:

```
 pub struct ResponseAttribute<'a, 'b> {
     pub required_attribute: RequiredAttribute,
     pub values: Cow<'b, [&'a str]>,
 }
```

File: src/idp/response_builder.rs [hunk 2]
```diff
@@ -49,19 +50,19 @@ fn build_attributes(formats_names_values: &[ResponseAttribute]) -> Vec<Attribute
...
             name: Some(attr.required_attribute.name.clone()),
             name_format: attr.required_attribute.format.clone(),
-            values: attr.values.iter().map(|value| {
-                AttributeValue {
+            values: attr
+                .values
+                .iter()
+                .map(|value| AttributeValue {
                     attribute_type: Some("xs:string".to_string()),
                     value: Some(value.to_string()),
```
Comment (hgzimmerman) [inline src/idp/response_builder.rs:58]: Aaah. This is silly. Why would they require `&'a str` in the first place, if they are just going to convert to String immediately after.

now I see what you were talking about. If `AttributeValue` also had lifetimes and `&'a str`, and it was converted to JSON before the lifetimes would expire, then sure I understand why they would do it this way.


`Vec<String>` is the way to go, forget about lifetimes.

Comment (jameswald) [reply src/idp/response_builder.rs:58]: Fixed with c3fc9f1b093bd7991aeb9dbc84e2c79eb64d208a.

**Comment by jameswald** [comment]
Rough week... getting sloppy. Replaced the merged commit with c3fc9f1b093bd7991aeb9dbc84e2c79eb64d208a.

File: src/idp/response_builder.rs (hunk 2 above)
Comment (jameswald) [reply src/idp/response_builder.rs:58]: Went with `Vec<String>` for the moment but we can discuss how we should ultimately solve this at https://github.com/spd-mfa/samael/commit/c3fc9f1b093bd7991aeb9dbc84e2c79eb64d208a#r112786776. Sounds like we'll want Cow + lifetimes after all.

**Pull request merged**